src/main/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java

```diff
@@ -78,7 +78,7 @@
             Pattern.compile("\\s*Violation Message Keys:\\s*");
 
     /** Regular expression for detecting ANTLR tokens(for e.g. CLASS_DEF). */
-    private static final Pattern TOKEN_TEXT_PATTERN = Pattern.compile("([A-Z]+_*)+[A-Z]+");
+    private static final Pattern TOKEN_TEXT_PATTERN = Pattern.compile("([A-Z]+_*+)+[A-Z]+");
```

Comment:
Why do we have `*` on `_`? Do we have any token with 2 or more underscores?
Shouldn't this be `([A-Z]+_)+[A-Z]+` ?
If there is a case with no underscore, then it should be something like  `([A-Z]+_)+?[A-Z]+` .

Reply:
to not fix too much problems in same time , I moved it to https://github.com/checkstyle/checkstyle/issues/9140